### ISSUE TITLE
InputControl (& related components) focus style doesn't support Windows High Contrast mode

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#50738]
### Description

Discovered while reviewing https://github.com/WordPress/gutenberg/pull/50609

It appears the focus style provided by `BackdropUI` [only provides a box-shadow style](https://github.com/WordPress/gutenberg/blob/6725f29434ebf538f3ba2440b5f83711a1283711/packages/components/src/input-control/styles/input-control-styles.tsx#L274-L276).

In Windows High Contrast mode box-shadows aren't visible. That's the reason why in most places we do provide a transparent outline, something along these lines:

```
// Windows High Contrast mode will show this outline, but not the box-shadow.
outline: 2px solid transparent;
```

`BackdropUI` should be updated to provide the transparent outline together with the blue box-shadow for the focus style.

### Step-by-step reproduction instructions

- Test on Windows.
- Enable Windows High Contrast mode
- Use the keyboard and move focus to any component that uses `BackdropUI`.
- Observe there's no focus style.
- Note: easiest way to test: check [the input control component in the storybook](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-inputcontrol--default).

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I was also able to confirm this issue on the StoryBook:

![storybook](https://github.com/WordPress/gutenberg/assets/54422211/0bddaa32-efea-4707-add7-c814b769c36d)

If this component is in the WordPress admin, it will be difficult to find the problem, because the outline will be overwritten by WP-Admin style (`forms.css`):

![select-control](https://github.com/WordPress/gutenberg/assets/54422211/e74fd649-a73a-4428-a29f-982a6a298c59)
